cscs-checks/apps/gromacs/gromacs_check.py

```diff
@@ -3,139 +3,223 @@
-
-        # Reset sources dir relative to the SCS apps prefix
-        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
-                                       'Gromacs', 'herflat')
-        self.keep_files = [output_file]
-
-        energy = sn.extractsingle(r'\s+Potential\s+Kinetic En\.\s+Total Energy'
-                                  r'\s+Conserved En\.\s+Temperature\n'
-                                  r'(\s+\S+){2}\s+(?P<energy>\S+)(\s+\S+){2}\n'
-                                  r'\s+Pressure \(bar\)\s+Constr\. rmsd',
-                                  output_file, 'energy', float, item=-1)
-        energy_reference = -3270799.9
-
-        self.sanity_patterns = sn.all([
-            sn.assert_found('Finished mdrun', output_file),
-            sn.assert_reference(energy, energy_reference, -0.001, 0.001)
-        ])
-
-        self.perf_patterns = {
-            'perf': sn.extractsingle(r'Performance:\s+(?P<perf>\S+)',
-                                     output_file, 'perf', float)
-        }
-
-        self.modules = ['GROMACS']
-        self.maintainers = ['VH', 'SO']
-        self.strict_check = False
-        self.use_multithreading = False
-        self.extra_resources = {
-            'switches': {
-                'num_switches': 1
-            }
+@rfm.simple_test
+class cscs_gromacs_check(gromacs_check):
+    modules = ['GROMACS']
+    maintainers = ['VH', 'VK']
+    use_multithreading = False
+    extra_resources = {
+        'switches': {
+            'num_switches': 1
         }
-        self.tags = {'scs', 'external-resources'}
-
-
-@rfm.parameterized_test(*([s, v]
-                          for s in ['small', 'large']
-                          for v in ['prod', 'maint']))
-class GromacsGPUCheck(GromacsBaseCheck):
-    def __init__(self, scale, variant):
-        super().__init__('md.log')
-        self.valid_systems = ['daint:gpu']
-        self.descr = 'GROMACS GPU check'
-        self.executable_opts = ['mdrun', '-dlb yes', '-ntomp 1', '-npme 0',
-                                '-s herflat.tpr']
-        self.variables = {'CRAY_CUDA_MPS': '1'}
-        self.num_gpus_per_node = 1
-        if scale == 'small':
-            self.valid_systems += ['dom:gpu']
-            self.num_tasks = 72
-            self.num_tasks_per_node = 12
-        else:
-            self.num_tasks = 192
-            self.num_tasks_per_node = 12
-
-        references = {
-            'maint': {
-                'large': {
-                    'daint:gpu': {'perf': (63.0, -0.10, None, 'ns/day')}
-                }
-            },
-            'prod': {
-                'small': {
-                    'dom:gpu': {'perf': (37.0, -0.05, None, 'ns/day')},
-                    'daint:gpu': {'perf': (35.0, -0.10, None, 'ns/day')}
-                },
-                'large': {
-                    'daint:gpu': {'perf': (63.0, -0.20, None, 'ns/day')}
-                }
+    }
+    executable_opts += ['-dlb yes', '-ntomp 1', '-npme -1']
```

Comment:
This line is not being executed...
The generated job script contains only the base class options:
```
srun gmx_mpi mdrun -nb gpu -s benchmark.tpr
```
It should be 
```
srun gmx_mpi mdrun -dlb yes -ntomp 1 -npme 0 -s herflat.tpr
```

Reply:
Oh, that's because #2268 !

Reply:
I will make it a post init hook with a FIXME note... When we fix #2268, then we revert to this version.
What do you think?

Reply:
It wouldn't work either because of #2268. As I have it now it won't work in any case, because the executable opts are set in a hook. I shouldn't have updated there. But there is a workaround if in the library we use `+=`. Can you try it?

Reply:
And generally, I think that libraries should always use `+=`...